### ISSUE TITLE
Missing braces in suggested code from `match_single_binding` lint

### ISSUE THREAD

**Issue by BBaoVanC** [rust-lang/rust-clippy#9575]
### Summary

In my example, I match the `ErrorKind` of an `io::Error`, but I only have one match arm. This causes clippy to warn about `match_single_binding`, which makes sense, but the suggestion it gives isn't valid code.

Clippy probably can't tell that `e.kind()` doesn't have any side effects so it tries to keep the call, but forgets to add braces.

### Reproducer

I tried this code:

```rust
use std::fs::File;

fn main() -> Result<(), String> {
    let _f = File::create("hello.txt").map_err(|e| match e.kind() {
        // TODO: match some other errors
        _ => String::from("misc i/o error"),
    })?;

    Ok(())
}
```

I expected to see this happen: the `clippy::match_single_binding` error to have valid code as a suggestion

Instead, this happened:

```
warning: this match could be replaced by its scrutinee and body
 --> src/main.rs:4:52
  |
4 |       let _f = File::create("hello.txt").map_err(|e| match e.kind() {
  |  ____________________________________________________^
5 | |         // TODO: match some other errors
6 | |         _ => String::from("misc i/o error"),
7 | |     })?;
  | |_____^
  |
  = note: `#[warn(clippy::match_single_binding)]` on by default
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#match_single_binding
help: consider using the scrutinee and body instead
  |
4 ~     let _f = File::create("hello.txt").map_err(|e| e.kind();
5 ~     String::from("misc i/o error"))?;
  |
```

The code shown under `help:` above isn't valid; it needs braces like this:

```rust
let _f = File::create("hello.txt").map_err(|e| {
    e.kind();
    String::from("misc i/o error")
})?;
```

Additionally, in the code above, the `e.kind();` is useless, but clippy might not be able to understand that.

### Version

```text
rustc 1.64.0 (a55dd71d5 2022-09-19)
binary: rustc
commit-hash: a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52
commit-date: 2022-09-19
host: x86_64-unknown-linux-gnu
release: 1.64.0
LLVM version: 14.0.6
```

### Additional Labels

@rustbot label +I-suggestion-causes-error
